lib/src/mapbox_maps_platform.dart

```diff
@@ -161,6 +161,23 @@ class _MapboxMapsPlatform {
     }
   }
 
+  Future<dynamic> addInteractionsListeners(
+      Interaction interaction, int interactionID) async {
+    try {
+      return _channel
+          .invokeMethod('interactions#add_interaction', <String, dynamic>{
```

Comment:
@evil159  - any better approach here you can think of?

Reply:
I think that's a good improvement -- I'll ticket it

Reply:
I tried this out in the latest commit with the creation of `_InteractionPigeon`. I'm curious to hear your thoughts on this. I think it cleans up the platform code just a bit.

Reply:
Hmmm... I don't see the reason to wrap interaction into `_InteractionPigeon` anymore(not sure why I didn't suggest it in the first place, sorry), we can just pass `_Interaction` itself, right?

Reply:
We can't pass `_Interaction` because it has types which are not supported by [MessageCodec](https://api.flutter.dev/flutter/services/StandardMessageCodec-class.html). Specifically: `FeaturesetDescriptor` and `_InteractionType`.

Reply:
I see, they've changed the way objects are encoded/decoded lately, you can make it work passing the `_Interaction` type to a pigeon-generated method, but yeah, it won't work out of box when using custom channels.